File: crossecore-java/src/java_/VisibilityKind.java
package java_;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public enum VisibilityKind implements org.eclipse.emf.common.util.Enumerator{
	NONE(1, "none", "none"),
	PUBLIC(2, "public", "public"),
	PRIVATE(3, "private", "private"),
	PROTECTED(4, "protected", "protected")
;	public static final int NONE_VALUE = 1;
	public static final int PUBLIC_VALUE = 2;
	public static final int PRIVATE_VALUE = 3;
	public static final int PROTECTED_VALUE = 4;
	
	private static final VisibilityKind[] VALUES_ARRAY =
		new VisibilityKind[] {
		NONE,
		PUBLIC,
		PRIVATE,
		PROTECTED
		};
	public static final List<VisibilityKind> VALUES = Collections.unmodifiableList(Arrays.asList(VALUES_ARRAY));
	
	public static VisibilityKind get(String literal) {
		for (int i = 0; i < VALUES_ARRAY.length; ++i) {
			VisibilityKind result = VALUES_ARRAY[i];
			if (result.toString().equals(literal)) {
				return result;
			}
		}
		return null;
	}
	
	public static VisibilityKind getByName(String name) {
		for (int i = 0; i < VALUES_ARRAY.length; ++i) {
			VisibilityKind result = VALUES_ARRAY[i];
			if (result.getName().equals(name)) {
				return result;
			}
		}
		return null;
	}
	
	public static VisibilityKind get(int value) {
		switch (value) {
			
			case NONE_VALUE: return NONE;
			
			case PUBLIC_VALUE: return PUBLIC;
			
			case PRIVATE_VALUE: return PRIVATE;
			
			case PROTECTED_VALUE: return PROTECTED;
		}
		return null;
	}
	private final int value;
	private final String name;
	private final String literal;
	
	private VisibilityKind(int value, String name, String literal) {
		this.value = value;
		this.name = name;
		this.literal = literal;
	}

	public int getValue() {
	  return value;
	}

	public String getName() {
	  return name;
	}

	public String getLiteral() {
	  return literal;
	}

	@Override
	public String toString() {
		return literal;
	}
			
}

File: crossecore-csharp/Java/Java/Java_/VisibilityKind.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	    public class VisibilityKind : EEnumerator
	    {
	    	
			public const int NONE_VALUE = 1;
			public const int PUBLIC_VALUE = 2;
			public const int PRIVATE_VALUE = 3;
			public const int PROTECTED_VALUE = 4;
			
			public static VisibilityKind NONE;
			public static VisibilityKind PUBLIC;
			public static VisibilityKind PRIVATE;
			public static VisibilityKind PROTECTED;
	
			private static VisibilityKind[] VALUES_ARRAY =
				new VisibilityKind[] {
				NONE, 
				PUBLIC, 
				PRIVATE, 
				PROTECTED
			};
	
	        public static VisibilityKind get(string literal)
	        {
	            for (int i = 0; i < VALUES_ARRAY.Length; ++i)
	            {
	                VisibilityKind result = VALUES_ARRAY[i];
	                if (result.ToString() == literal)
	                {
	                    return result;
	                }
	            }
	            return null;
	        }
	
	        public static VisibilityKind getByName(string name)
	        {
	            for (int i = 0; i < VALUES_ARRAY.Length; ++i)
	            {
	                VisibilityKind result = VALUES_ARRAY[i];
	                if (result.getName()==name)
	                {
	                    return result;
	                }
	            }
	            return null;
	        }
	
	        public static VisibilityKind get(int value)
	        {
	            switch (value)
	            {
				case NONE_VALUE: return NONE;
				case PUBLIC_VALUE: return PUBLIC;
				case PRIVATE_VALUE: return PRIVATE;
				case PROTECTED_VALUE: return PROTECTED;
	            }
	            return null;
	        }
	
	        private int value;
	        private string name;
		    private string literal;
	
	        private VisibilityKind(int value, string name, string literal)
	        {
	            this.value = value;
	            this.name = name;
	            this.literal = literal;
	        }
	
	        public string getLiteral()
	        {
	            return literal;
	        }
	
	        public string getName()
	        {
	            return name;
	        }
	
	        public int getValue()
	        {
	            return value;
	        }
	    }
}
